### PR TITLE
[DDO-2885] Migrate local development config out of firecloud-develop

### PR DESCRIPTION
**Pull request by choover-broad**
**Ticket**: [DDO-2885](https://broadworkbench.atlassian.net/browse/DDO-2885)

The goal of this PR is to do **the minimal amount of work** to migrate the existing Rawls local development configuration out of firecloud-develop.

Under the existing process, developers clone firecloud-develop and run `sh run-context/local/scripts/firecloud-setup.sh`, which populates the Rawls repo's `configs` subdirectory with a bunch of configuration files and secrets, as well as a script called `docker-rsync-local-rawls.sh`, which can be run to spin up a local Rawls server inside a container.

This PR replaces the firecloud-develop step with a new helper script, `local-dev/bin/render`, that **generates the same set of files**. The [`render` tool](https://github.com/broadinstitute/firecloud-develop-shim) generates 3 types of configuration files, giving it approximate parity with firecloud-develop's `configure.rb`:
* secrets pulled from Vault (declared in `local-dev/secrets.yaml`)
* flat configuration files (checked in under `local-dev/templates` without a `.ctmpl`  extension)
* consul templates (checked in under `local-dev/templates` with a `.ctmpl` extension)

### Testing

I verified that I could run the `docker-rsync-local-rawls.sh` script, connect to Rawls swagger at https://local.broadinstitute.org:20443/, authorize to Swagger via the implicit OAuth flow, and successfully execute the `GET /api/workspaces` API call. 

### Modified files

A diff on the files generated by the old process and the new process showed the following delta:

#### changed: `billing-account.pem`

The firecloud-develop configs pull an old static billing service account key located at `secret/dsde/firecloud/dev/common/billing-account.json`; this PR pulls a key that is automatically rotated by Yale and stored in Vault at `secret/dsde/firecloud/dev/common/billing-account-yale.pem`.

####  changed:`rawls.conf`

Single-line removal of a comment that no longer made sense with the new rawls.conf.ctmpl template in this PR.
```
diff -B -w -r config/rawls.conf config-from-fc-dev/rawls.conf
4a5
>   # This billing account ID corresponds to the billing account 'Broad Institute - 8201528'
```

#### added: `server.crt`, `server.key`

These files are the `local.broadinstitute.org` cert & key pair located at `secret/dsde/firecloud/local/common`. They were removed from Rawls' firecloud-develop configs by https://github.com/broadinstitute/firecloud-develop/pull/3326, which broke TLS for local Rawls. This PR adds them back.

#### changed: `site.conf`

Simplify the Apache proxy's OAuth claim allowlist so that it includes just the legacy dev Google OAuth credential and the Terra dev B2C app id, which are used by the Rawls swagger page for auth.

#### deleted: `tcell_agent.config`

There's no need to run TCell locally.

### Related

* https://github.com/broadinstitute/firecloud-develop/pull/3379 will be merged after this PR to prevent confusion (i.e. people updating old files that are no longer used)
---

**PR checklist**

- [ ] Include the JIRA issue number in the PR description and title
- [ ] Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] If you changed anything in `model/`, then you should [publish a new official `rawls-model`](https://github.com/broadinstitute/rawls/blob/develop/README.md#publish-rawls-model) and update `rawls-model` in [Orchestration's dependencies](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/project/Dependencies.scala).
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green, including CI tests
- [ ] **Squash commits and merge** to develop (branches are automatically deleted after merging)
- [ ] Inform other teams of any substantial changes via Slack and/or email

[DDO-2885]: https://broadworkbench.atlassian.net/browse/DDO-2885?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: local-dev/templates/mod_security_ignore.conf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+SecRule REQUEST_URI "@contains importEntities" phase:1,id:'1000000001',nolog,allow,ctl:ruleEngine=Off,ctl:auditEngine=Off
+SecRule REQUEST_URI "@contains batchUpsert" phase:1,id:'1000000002',nolog,allow,ctl:ruleEngine=Off,ctl:auditEngine=Off
+SecRule REQUEST_URI "@contains loading..." phase:1,id:'1000000003',nolog,allow,ctl:ruleEngine=Off,ctl:auditEngine=Off
```
Comment (davidangb) [inline local-dev/templates/mod_security_ignore.conf:3]: where did these three entries come from?

File: local-dev/templates/rawls.conf.ctmpl [hunk 2]
```diff
@@ -0,0 +1,460 @@
...
+{{- $rawlsOauthCredential := (secret "secret/dsde/firecloud/dev/rawls/rawls-oauth-credential.json").Data }}
+{{- $rawlsConf := (secret "secret/dsde/firecloud/dev/rawls/rawls.conf").Data -}}
+{{- $refreshTokenCredential := (secret "secret/dsde/firecloud/dev/common/refresh-token-oauth-credential.json").Data -}}
+{{- $rawlsSaKey := (secret "secret/dsde/firecloud/dev/rawls/rawls-account.json").Data -}}
+{{- $cromwellSaKey := (secret "secret/dsde/firecloud/dev/cromwell/cromwell-account.json").Data -}}
+{{- $staticPerimeterProjects := (secret "secret/dsde/firecloud/dev/rawls/service-perimeters/static-projects").Data -}}
+{{- $b2cAppId := (secret "secret/dsde/terra/azure/dev/b2c/application_id" ).Data.value -}}
+{{- $landingZonePostgres := (secret "secret/dsde/terra/azure/dev/workspacemanager/azure-postgres-credential" ).Data -}}
+
+backRawls = true
```
Comment (davidangb) [inline local-dev/templates/rawls.conf.ctmpl]: for local development, this should be false by default. I think that's how we have it in firecloud-develop.

File: local-dev/templates/mod_security_ignore.conf (hunk 1 above)
Comment (choover-broad) [reply local-dev/templates/mod_security_ignore.conf:3]: Thanks for taking look! I created this PR by working backwards - render configuration from firecloud-develop using the [firecloud-setup.sh](https://github.com/broadinstitute/firecloud-develop/blob/dev/run-context/local/scripts/firecloud-setup.sh) script, then reconstruct the output in the Rawls repo with secrets pulled from Vault. This template comes from [common templates](https://github.com/broadinstitute/firecloud-develop/blob/dev/base-configs/common/mod_security/mod_security_ignore.conf). I'm happy to remove it (and anything else in this PR) that seems unnecessary.

File: local-dev/templates/rawls.conf.ctmpl (hunk 2 above)
Comment (choover-broad) [reply local-dev/templates/rawls.conf.ctmpl]: @davidangb When I run `./run-context/local/scripts/firecloud-setup.sh` and select `rawls` (as described in the README), it produces a `rawls.conf` with `backRawls = true`. What process do you use generate the config templates? Maybe I'm missing some additional local config?

Comment (choover-broad) [reply local-dev/templates/rawls.conf.ctmpl]: ah, I figured it out - the docker script overrides it via system property: https://github.com/broadinstitute/rawls/pull/2383/files#diff-ae2e558e0e238c2247080ce69bad049968b7643f2411424256631f556489f45aR74

In any case, I'll adjust the template so it's less confusing

Comment (davidangb) [reply local-dev/templates/rawls.conf.ctmpl]: I'm also seeing a difference between `firecloud-setup.sh` and `render-configs.sh`. In the latter, the INSTANCE_TYPE is specified at render-time, without a default:
https://github.com/broadinstitute/firecloud-develop/blob/9710fbbada766e39327cda92c9df09a7ffb8ac0a/render-configs.sh#L43-L44

and with that said, I think the default should be `backRawls=false` and developers can override that if they want to (don't need to add it to the render script; devs can just edit the rendered config file). Perhaps we can get opinions from other rawls devs on this too.

**Review by aherbst-broad — APPROVED** [review]
Smoke tested this locally, I was able to list and get workspaces. Changes look reasonable otherwise, thanks for doing this.

**Review by davidangb — APPROVED** [review]
thank you!!

**Pull request merged**